test/claim.test.ts

```diff
@@ -114,37 +111,179 @@ describe('StakingPool.claim', () => {
-      const userDataAfter = await getUserData(testEnv, alice, firstRound);
+      const poolDataAfter = await getPoolData(testEnv);
+      const userDataAfter = await getUserData(testEnv, alice);
 
       expect(poolDataAfter).to.be.equalPoolData(expectedPoolData);
       expect(userDataAfter).to.be.equalUserData(expectedUserData);
     });
+
+    // TODO if pool is colosed reward is not increased
+    
+    // it('if pool is closed, reward is not increased', async () => {
+    //   await testEnv.stakingPool.connect(alice).stake(amount.mul(2));
+
+    //   const poolDataBefore = await getPoolData(testEnv);
+    //   const userDataBefore = await getUserData(testEnv, alice);
+      
+    //   const [expectedPoolData, expectedUserData] = expectDataAfterClaim(
+    //     poolDataBefore,
+    //     userDataBefore,
+    //     secondRoundStartTimestamp
+    //   );
+
+    //   const tx = await testEnv.stakingPool.connect(deployer).closePool();
+    //   await advanceTimeTo(await getTimestamp(tx), secondRoundStartTimestamp);
+    //   await testEnv.stakingPool.connect(alice).claim();
+
+    //   const poolDataAfter = await getPoolData(testEnv);
+    //   const userDataAfter = await getUserData(testEnv, alice);
+
+    //   expect(poolDataAfter).to.be.equalPoolData(expectedPoolData);
+    //   expect(userDataAfter).to.be.equalUserData(expectedUserData);
+    // });
+
+    context('change rewardPerSecond', async () => {
+      beforeEach('init the pool and stake in pool', async () => {
+        await testEnv.rewardAsset.connect(deployer).faucet();
+        await testEnv.rewardAsset.connect(deployer).approve(testEnv.stakingPool.address, RAY);
+        await testEnv.stakingPool
+          .connect(deployer)
+          .initNewPool(
+            rewardPersecond,
+            firstRoundStartTimestamp,
+            duration
+          );
+        await testEnv.stakingAsset.connect(alice).faucet();
+        const tx = await testEnv.stakingAsset.connect(alice).approve(testEnv.stakingPool.address, RAY);
+        await advanceTimeTo(await getTimestamp(tx), firstRoundStartTimestamp);
+        await testEnv.stakingPool.connect(alice).stake(amount);
+      });
+
+      it('rewardPerSecond is changed and claim', async () => {
+        const tx = await testEnv.stakingPool.connect(deployer).inputNextReward(inputAmount); 
+
+        const poolDataBefore = await getPoolData(testEnv);
+        const userDataBefore = await getUserData(testEnv, alice);
+        await advanceTimeTo(await getTimestamp(tx), passTimestamp);
+  
+        const [expectedPoolData_1, expectedUserData_1] = expectDataAfterStake(
+          poolDataBefore,
+          userDataBefore,
+          passTimestamp,
+          BigNumber.from(0)
+        );
+        
+        const claimTx = await testEnv.stakingPool.connect(alice).claim();
+        const [expectedPoolData_2, expectedUserData_2] = expectDataAfterClaimSetReward(
+          expectedPoolData_1,
```

Comment:
Why does this function get `expectedPoolData_1` and `expectedUserData_1`? I think it should take actual pool data from the current contract state as parameters.